### PR TITLE
column computation speed up #13

### PR DESCRIPTION
**Pull request by vdbijl**
Improve speed of the computation of the column position. We keep track
of the starting position of the current line, while lexing.
The speed up helps especially with large files. The lexing of our
test-file went from 10s to 2s.

### CODE REVIEW

File: lib/oedipus_lex.rb [hunk 1]
```diff
@@ -147,7 +147,7 @@ def to_ruby state, predicates, exclusive
 
       cond = if exclusive or not start_state then
                check
-             elsif start_state =~ /^:/ then
+             elsif /^:/.match?(start_state) then
```
Comment (vdbijl) [inline lib/oedipus_lex.rb:150]: While looking through the code I saw this optimization. I'm ok with removing it if you do not like it.

File: lib/oedipus_lex.rb [hunk 2]
```diff
@@ -359,7 +359,7 @@ def lex_groupend start_state, regexp, action = nil
   ##
   # Process a +state+ lexeme.
 
-  def lex_state new_state
+  def lex_state _new_state
```
Comment (vdbijl) [inline lib/oedipus_lex.rb:362]: minor, seems like the parameter is not used, but I'm ok if you do not like it.

File: lib/oedipus_lex.rb [hunk 3]
```diff
@@ -474,12 +474,18 @@ def action
 
         attr_accessor :old_pos
 
+        ##
+        # The position of the start of the current line. Only available if the
+        # :column option is on.
+
+        attr_accessor :start_of_current_line_pos
```
Comment (vdbijl) [inline lib/oedipus_lex.rb:481]: keep track of the start of the current line. We use this as the off-set of the column position computation.

File: lib/oedipus_lex.rb [hunk 4]
```diff
@@ -556,7 +565,13 @@ def next_token
 
           until ss.eos? or token do
 % if option[:lineno] then
```
Comment (vdbijl) [inline lib/oedipus_lex.rb:567]: lineno and column options interfere. I assumed column also assumes lineno, so I added it inside the lineno option

**Comment by vdbijl** [comment]
This is my first little pull request on github, so my apologies if I did something wrong. No problem to adjust the code.

**Comment by zenspider** [comment]
I split this PR up into your oedipus_lex.rb change and then the versionbump+bootstrap (so they match) & just released it.

Thank you!

**Pull request closed without merging**